src/openforms/submissions/api/validators.py

```diff
@@ -33,6 +33,11 @@ def __call__(self, data: dict, field: JSONField):
             if not (component_key := component.get("key")):
                 continue
 
+            # in case the component or its parent component is hidden the key will not be
+            # part of the data.
+            if component_key not in data:
+                continue
```

Comment:
What happens now if a field is prefilled but not hidden and the data is omitted from the submission data? It kinda looks now like a user can clear a prefilled field that is read-only.